internal/tofu/context_apply2_test.go

```diff
@@ -1540,41 +1540,42 @@ output "out" {
 
 	_, diags = ctx.Plan(context.Background(), m, state, opts)
 	assertNoErrors(t, diags)
-	return
 	// TODO: unreachable code
-	otherProvider.ConfigureProviderCalled = false
-	otherProvider.ConfigureProviderFn = func(req providers.ConfigureProviderRequest) (resp providers.ConfigureProviderResponse) {
-		// check that our config is complete, even during a destroy plan
-		expected := cty.ObjectVal(map[string]cty.Value{
-			"local":  cty.ListVal([]cty.Value{cty.StringVal("first-ok"), cty.StringVal("second-ok")}),
-			"output": cty.ListVal([]cty.Value{cty.StringVal("first-ok"), cty.StringVal("second-ok")}),
-			"var": cty.MapVal(map[string]cty.Value{
-				"a": cty.StringVal("first"),
-				"b": cty.StringVal("second"),
-			}),
-		})
+	if 1 == 0 {
+		otherProvider.ConfigureProviderCalled = false
+		otherProvider.ConfigureProviderFn = func(req providers.ConfigureProviderRequest) (resp providers.ConfigureProviderResponse) {
+			// check that our config is complete, even during a destroy plan
+			expected := cty.ObjectVal(map[string]cty.Value{
+				"local":  cty.ListVal([]cty.Value{cty.StringVal("first-ok"), cty.StringVal("second-ok")}),
+				"output": cty.ListVal([]cty.Value{cty.StringVal("first-ok"), cty.StringVal("second-ok")}),
+				"var": cty.MapVal(map[string]cty.Value{
+					"a": cty.StringVal("first"),
+					"b": cty.StringVal("second"),
+				}),
+			})
 
-		if !req.Config.RawEquals(expected) {
-			resp.Diagnostics = resp.Diagnostics.Append(fmt.Errorf(
-				`incorrect provider config:
+			if !req.Config.RawEquals(expected) {
+				resp.Diagnostics = resp.Diagnostics.Append(fmt.Errorf(
+					`incorrect provider config:
 expected: %#v
 got:      %#v`,
-				expected, req.Config))
-		}
+					expected, req.Config))
+			}
 
-		return resp
-	}
+			return resp
+		}
 
-	opts.Mode = plans.DestroyMode
-	// skip refresh so that we don't configure the provider before the destroy plan
-	opts.SkipRefresh = true
+		opts.Mode = plans.DestroyMode
+		// skip refresh so that we don't configure the provider before the destroy plan
+		opts.SkipRefresh = true
 
-	// destroy only a single instance not included in the moved statements
-	_, diags = ctx.Plan(context.Background(), m, state, opts)
-	assertNoErrors(t, diags)
+		// destroy only a single instance not included in the moved statements
+		_, diags = ctx.Plan(context.Background(), m, state, opts)
+		assertNoErrors(t, diags)
 
-	if !otherProvider.ConfigureProviderCalled {
-		t.Fatal("failed to configure provider during destroy plan")
+		if !otherProvider.ConfigureProviderCalled {
+			t.Fatal("failed to configure provider during destroy plan")
+		}
```

Comment:
I think we should just delete all of this if we can't make it actually runnable. Unreachable test code isn't going to help us catch any problems. :shrug:

Reply:
I left this as AFAIK it represents a path that might represent a bug that was known but ignored in the past.